collections/gamesCollection.mjs

```diff
@@ -45,21 +47,17 @@ class GamesCollection {
 
     url += "?format=json";
 
-    this.yf.api(this.yf.GET, url, (e, data) => {
-      if (e) {
-        return cb(e);
-      }
-
-      const games = parseCollection(data.fantasy_content.games, subresources);
-      return cb(null, games);
-    });
+    return this.yf.api(this.yf.GET, url)
+      .then(data => parseCollection(data.fantasy_content.games, subresources))
+      .then(games => cb(null, games))
+      .catch(e => cb(e));
```

Comment:
The general pattern for most invocations of yf.api was to take the "raw data => mapped object" portion of the callback and run that after yf.api returns. Then always invoke the callback when the promise resolves with the data, and invoke it with the error when it rejects.

Reply:
I also considered adding a mappingfunc parameter to yf.api a and letting yf.api do all the callback invoking. It would result in less changes (and total code) in each of these resources, but I worried it was further from the current API.